plot-lib/plot/private/common/ticks.rkt

```diff
@@ -35,10 +35,16 @@
 
 (:: ticks-generate (-> ticks Real Real (Listof tick)))
 (define (ticks-generate t x-min x-max)
-  (match-define (ticks layout format) t)
+  (match-define (ticks layout fmt) t)
   (define ts (map pre-tick-inexact->exact (layout x-min x-max)))
   (match-define (list (pre-tick #{xs : (Listof Real)} #{majors : (Listof Boolean)}) ...) ts)
-  (map tick xs majors (format x-min x-max ts)))
+  (define labels (fmt x-min x-max ts))
+  (unless (= (length labels) (length ts))
+    (error
```

Comment:
Since `ticks-generate` is public, the error should use its name. Suggestion:
```
(raise-arguments-error
  'ticks-generate
  "ticks-format must return one label for each pre-tick"
  "num. expected" (length ts)
  "num. given" (length labels)
  "labels" labels
  "ticks-layout" layout
  "ticks-format" fmt)
```